mitodl/mit-open
# Collect ui components to ol-design package

# What are the relevant tickets?
#161 

# Description (What does it do?)
See #161. This PR moves many presentation components to a new `ol-design` package and removes the `ol-forms` package. I removed some components that were completely unused or that can be replaced with builtins. 

Also, no package directly imports from `@mui/material`. Instead, `ol-design` re-exports (some of) `@mui/material. The rational for this was:

> 1. `@mui/material` is **too** customizable, leading to inconsistent styling/behavior throughout pages. By defining our own wrappers, we can expose the customizability we want, but enforce consistency in other options.
> 2. If we decide to swap `@mui/material` for another component library, we have fewer places in our own code that needs to change.

The first goal ended up not really being accomplished in this PR: I decided to directly re-export many components (e.g., Button; TextField, many others) directly from MUI. Thus, their interfaces aren't really constrained. Re-exporting had the advantage of preserving MUI's nice docstrings and also means that ref forwarding works naturally without needing to wrap components in `forwardRef`. 

# How can this be tested?
Besides CI tests passing, I would check:
1. Homepage looks same. *There is a slight difference in the chips below searchbar. But I think this difference is fine for now since we don't have final designs.*
2. Check http://localhost:8063/learningpaths/ *Need to be logged in as admin to create new paths.*
    - creating / editing paths should work
    - Check the "..." menu for editing paths. The "..." menu is new.
    -  Check the Home button new 404 page (e.g., http://localhost:8063/learningpaths/fake_id) which uses new ButtonLink component
